### DISCUSSION TITLE
How to make a custom drone?

### DISCUSSION THREAD

**Discussion by oestby** [microsoft/AirSim#838] [Support Q&A]
Hi, I wanted to know if it is possible to make a custom drone with AirSim?

The AirSim drone is larger than ours, its lighter, it has a different center of mass, and a different thrust.

Is there some parameters where we can just write in these values so that the built-in drone behaves with updated physics? 

We also need to change the UnrealEngine drone model with our own 3D mesh. Where can we do this?

Sincerely

**Comment by sytelus** [comment]
You can create 3D mesh and pawn BP and use this setting to replace default one: https://github.com/Microsoft/AirSim/blob/v1.1.10/docs/settings.md#pawnpaths

You can create new config like this for physics parameters: https://github.com/Microsoft/AirSim/blob/v1.1.10/AirLib/include/vehicles/multirotor/configs/SimpleFlightQuadX.hpp

**Comment by AloshkaD** [comment]
@Oskiboy how did creating a custom drone go? I'm looking into building one myself for simulating the jetson Tx2 board I have on board a larger drone than the standard drone in airsim.

**Comment by oestby** [comment]
@zungam is the one who fixed this!
Sent him a message, so he'll see this.

**Comment by zungam** [comment]
Im sorry, we couldn't make it due to time constraints.

**Comment by AloshkaD** [comment]
@Oskiboy @zungam Thanks guys!

**Comment by Ischio** [comment]
Did someone managed to do it?
I managed to modify masses, lengths and thust/toque parameters according to a real drone but the drone can't fly properly or, with other data, it doesn't take off. 
I think the problem is my drone has a mass of 0.087 (Weight I suppose, but in the SimpleFlightQuadXParameters is written mass ) while the standard one is 1 and it's written it can be modified from 0.8 to 1.6 .
Does it means I can't simulate my drone??
Thank you

**Comment by chaitanyantr** [comment]
@Ischio By any chance you were able to fly the drone by changing parameters.Can you please share your experience on that.

Thankyou,
Chaitanya krishna

**Comment by EdwinMeriaux** [comment]
@sytelus How would we create 3D mesh and Pawn BP?

**Comment by Dave-Poissant** [comment]
Hi, I've tryed following the directions here:
https://github.com/Microsoft/AirSim/blob/v1.1.10/docs/settings.md#pawnpaths
by changing the "DefaultQuadrotor" pawn to my custom one.

But when trying to play the sim, I've got an exception thrown here:
https://github.com/microsoft/AirSim/blob/master/Unreal/Plugins/AirSim/Source/Vehicles/Multirotor/FlyingPawn.cpp#L21
by the UE4Editor-Engine.dll saying that it perceives an access violation reading location "whatever the adress". I've tryed running VS in admin, but still facing the problem.

Now my guess would be that the initializer is trying to load some camera actors out of my custom drone pawn, but its facing a nullptr. If so, I think I would need to implement some child actors cameras into my custom drone .uasset but I'm fairly new to unreal and AirSim particularly and I'm not sure if I'm making a correct assessment.

Anyone has ever faced this exception or would have an idea of how to fix it?
